src/preview/topics.rs

```diff
@@ -0,0 +1,315 @@
+/// This will run more or less forever and yield items as long as you're
+/// subscribed and someone is publishing.
+pub struct Subscription {
+    inner: tonic::Streaming<pubsub::SubscriptionItem>,
+}
+
+impl Subscription {
+    /// Wait for the next item in the stream.
+    ///
+    /// Result::Ok(Some(item))    -> the server sent you a subscription item!
+    /// Result::Ok(None)          -> the server is done - there will be no more items!
+    /// Result::Err(MomentoError) -> something went wrong - log it and maybe reach out if you need help!
+    pub async fn item(&mut self) -> Result<Option<SubscriptionItem>, MomentoError> {
+        self.inner
+            .message()
+            .await
+            .map_err(|e| e.into())
+            .map(Subscription::map_into)
+    }
+
+    /// Yeah this is a pain, but doing it here lets us yield a simpler-typed subscription stream.
+    /// Also, we don't want to expose protocol buffers types outside of the sdk, so some type map
+    /// had to happen. It's all one-off at the moment though so might as well leave it as one
+    /// triangle expression =)
+    fn map_into(possible_item: Option<pubsub::SubscriptionItem>) -> Option<SubscriptionItem> {
+        match possible_item {
+            Some(item) => match item.kind {
+                Some(kind) => match kind {
+                    pubsub::subscription_item::Kind::Item(item) => match item.value {
+                        Some(value) => {
+                            let sequence_number = item.topic_sequence_number;
+                            match value.kind {
+                                Some(topic_value_kind) => {
+                                    Some(SubscriptionItem::Value(SubscriptionValue {
+                                        topic_sequence_number: sequence_number,
+                                        kind: match topic_value_kind {
+                                            pubsub::topic_value::Kind::Text(text) => {
+                                                ValueKind::Text(text)
+                                            }
+                                            pubsub::topic_value::Kind::Binary(binary) => {
+                                                ValueKind::Binary(binary)
+                                            }
+                                        },
+                                    }))
+                                }
+                                // Broken protocol
+                                None => Some(SubscriptionItem::Discontinuity(Discontinuity {
+                                    last_sequence_number: None,
+                                    new_sequence_number: sequence_number,
+                                })),
+                            }
+                        }
+                        None => None, // Broken protocol
+                    },
+                    pubsub::subscription_item::Kind::Discontinuity(_) => todo!(),
```

Comment:
Is there some complexity here I'm not seeing? You already have the discontinuity types, isn't it a fairly straightforward conversion for this section? Any reason not to include it now?

Reply:
I missed this arm, good catch. It's addressed now.